kotlin/viewregistry-android/src/main/java/com/squareup/viewregistry/ModalContainer.kt

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.viewregistry
+
+import android.app.Dialog
+import android.content.Context
+import android.support.annotation.StyleRes
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.BehaviorSubject
+import kotlin.reflect.jvm.jvmName
+
+/**
+ * Base class for containers that show [HasModals.modals] in [Dialog]s. [forAlertContainerScreen]
+ * and [forContainerScreen] should cover many needs, and subclasses are simple to create.
```

Comment:
Not obvious to me how `forAlertContainerScreen` and `forContainerScreen` are related to this class, is the kdoc missing a sentence?

Reply:
They're companion methods on it. I'll tweak the language to call them factory methods explicitly.

Reply:
```
/**
 * Base class for containers that show [HasModals.modals] in [Dialog]s. 
 * 
 * The concrete implementations returned by the factory methods [forAlertContainerScreen] 
 * and [forContainerScreen] should cover many specific needs, and where those are too 
 * limiting subclasses are simple to create.
 */
```